src/Spago/Command/Fetch.purs

```diff
@@ -109,12 +140,28 @@ run { packages, ensureRanges, isTest } = do
 
   -- if the flag is selected, we kick off the process of adding ranges to the config
   when ensureRanges do
-    { configPath, yamlDoc } <- getPackageConfigPath
+    { configPath, yamlDoc, transitiveDeps } <- case selectedPackage of
+      SinglePackage p -> pure p
+      AllWorkspacePackages _ -> die
+        [ "No package found in the root configuration."
+        , "Please use the `-p` flag to select a package to which to add ranges."
+        ]
```

Comment:
This error message was packaged in a single function and is now repeated three times, could we reintroduce that?

Reply:
The reason why I split this up was because the error message was off for the second case:
1. Error message makes sense here because we're installing something: https://github.com/purescript/spago/pull/1018/files#diff-4529e47b55cd312f9954aa4027ac032176d4c22848d7e5ccf2c7adb942587e38L99
2. but not here because we could NOT pass the `install` flag and instead just be doing the ensure ranges. In that situation, why would the error mention anything about installation? https://github.com/purescript/spago/pull/1018/files#diff-4529e47b55cd312f9954aa4027ac032176d4c22848d7e5ccf2c7adb942587e38L112

Oh, and where is the third time it's being duplicated?

Reply:
Lines 104 and 128 are the same messages with different structures, 146 is a slightly different but with the same structure as 128.

I guess this is alright, but we should at least collect the shared error message in one place

Reply:
Hm.... I looked at sharing the error message, but then came up with a different idea. What if we defined the current package install block of code and 'ensureRanges' block of code into let-bound functions and then just call those functions in their respective places?

Reply:
I'm not sure how that would look like, but we can try it out 😄

Reply:
I think there's a bug in the current PR. If I have a package in the workspace `spago.yaml`, and don't select a package BUT I do have the `ensureRanges` flag, then the the root package doesn't get ranges added to it. I believe this occurs because I only mark the root package as a `SinglePackage` if the `packages` to install is not null. However, I don't also check whether the `ensureRanges` value is true.

Reply:
Ah I see - can we write a test to verify this?

Reply:
I still need to add a test, but see latest commits for what I did here.

Reply:
Hm... This still seems to differ from the current behavior. Sigh...

Reply:
I've updated the code back to the original approach here. Seems simpler at the end of the day.